src/ConfigEditor/ConfigEditor.tsx

```diff
@@ -57,21 +67,44 @@ export function ConfigEditor(props: Props) {
     const res: Secret = await getBackendSrv().post(resourcesURL + '/secret', { secretARN: arn });
     return res;
   };
+  const { onOptionsChange: propsOnOptionsChange } = props;
   useEffect(() => {
     if (arn) {
       fetchSecret(arn).then((s) => {
-        props.onOptionsChange({
-          ...props.options,
-          jsonData: {
-            ...props.options.jsonData,
-            clusterIdentifier: s.dbClusterIdentifier,
-            dbUser: s.username,
-          },
+        getClusterUrl(s.dbClusterIdentifier).then((url) => {
+          propsOnOptionsChange({
+            ...props.options,
+            url,
+            jsonData: {
+              ...props.options.jsonData,
+              clusterIdentifier: s.dbClusterIdentifier,
+              dbUser: s.username,
+            },
+          });
         });
       });
     }
   }, [arn]);
 
+  // Clusters
+  const fetchClusters = async () => {
+    const res: Cluster[] = await getBackendSrv().get(resourcesURL + '/clusters');
+    return res.map((c) => ({
+      label: c.clusterIdentifier,
+      value: c.clusterIdentifier,
+      description: `${c.endpoint.address}:${c.endpoint.port}`,
+    }));
+  };
+
+  const getClusterUrl = async (clusterID: string) => {
+    const { jsonData } = props.options;
+    if (clusterID != jsonData.clusterIdentifier) {
+      const clusters = await fetchClusters();
+      return `${clusters.find((c) => c.value === clusterID)?.description || clusterID}/${jsonData.database || ''}`;
+    }
+    return props.options.url;
+  };
```

Comment:
I tried to use memoize here to avoid fetching unnecessary clusters (if clusterID is still the same) but I could not get it to work so I added the memo check myself 😉